Adyen/Core/Core Protocols/PresentableComponent.swift

```diff
@@ -69,8 +69,16 @@ public protocol TrackableComponent: Component {
 @_spi(AdyenInternal)
 extension TrackableComponent where Self: PaymentMethodAware {
 
-    public func sendTelemetryEvent() {
+    public func sendTelemetryEvent(withAmount amount: Amount?) {
         let flavor: TelemetryFlavor = _isDropIn ? .dropInComponent : .components(type: paymentMethod.type)
-        context.analyticsProvider.sendTelemetryEvent(flavor: flavor)
+        context.analyticsProvider.sendTelemetryEvent(flavor: flavor, amount: amount)
+    }
+}
+
+@_spi(AdyenInternal)
+extension TrackableComponent where Self: PaymentAware & PaymentMethodAware {
+    
+    public func sendTelemetryEvent() {
+        sendTelemetryEvent(withAmount: payment?.amount)
```

Comment:
If you added the `PaymentAware` check here, then I think you don't need to re-conform to the function on all the below places